aiida/restapi/translator/data/structure.py

```diff
@@ -165,10 +209,19 @@ def get_visualization_data(node, format=None, supercell_factors=[1, 1, 1]):
                 }
 
             # These will be passed to ChemDoodle
-            response = {"s": [cell_json],
+            response["str_viz_info"]["data"] = {"s": [cell_json],
                             "m": [{"a": atoms_json}],
                             "units": '&Aring;'
                             }
+            response["str_viz_info"]["format"] = "default (chemDoodle)"
+
+        # Add extra information
+        pbc = node.pbc
+        info = get_extension(node.cell, pbc)
+        response["dimensionality"] = info["dim"]
+        response[info["label"]] = info["value"]
+        response["pbc"] = pbc
+        response["formula"] = node.get_formula()
```

Comment:
Is there any function available in aiida to get unit? or is it "Angstrom^dimension" for all?

Reply:
I think for the cell it's safe to assume that the units are Angstroms.
@giovannipizzi correct?

Reply:
So, if it's a cell from a StructureData, yes. If we want to keep this as a generic function, it's not the best idea to append the dimension as 'angstrom' by default. I think that if you pass the cell, you know its units, and since you have the 'dim' variable in the retdict, the units can be created in the frontend. Actually, this makes things easier, as otherwise in the frontend you have to convert the string angstrom to the proper symbol, and ^2 to a superscript.

If appropriate, I would suggest:
- convert the get_dimensionality() function as a method of StructureData (in this way, you can just use `self.cell` and `self.pbc` without the need of function parameters).
- remove the units (in the case of a method of StructureData, on the other hand, they could be useful to have, but still I'm not sure of how you would use it in the frontend without a big effort... if you really want to keep it, units are always lowercase: "angstrom"). But what I would do is just to document the units in the docstring (i.e., angstrom^dim) but not return it in the dictionary
- as we know we are using structuredata, hardcode the units in the frontend nicely as Å and the dim as superscript (and add a documentation for the API documenting the units the API endpoint will return)